contracts/staking/ValidatorManager.sol

```diff
@@ -110,56 +110,33 @@ abstract contract ValidatorManager is Initializable, ContextUpgradeable, IValida
     ) external {
         ValidatorManagerStorage storage $ = _getValidatorManagerStorage();
         require(!$._initializedValidatorSet, "ValidatorManager: already initialized validator set");
-
-        WarpMessage memory warpMessage = _getPChainWarpMessage(messageIndex);
-        // Parse the Warp message into SubnetConversionMessage
-        bytes32 subnetConversionID =
-            ValidatorMessages.unpackSubnetConversionMessage(warpMessage.payload);
-
-        // Verify that the sha256 hash of the Subnet conversion data matches with the Warp message's subnetConversionID.
-        bytes memory encodedConversion = abi.encodePacked(
-            subnetConversionData.convertSubnetTxID,
-            subnetConversionData.blockchainID,
-            subnetConversionData.validatorManagerAddress
-        );
-        uint256 numInitialValidators = subnetConversionData.initialValidators.length;
-        for (uint256 i; i < numInitialValidators; i++) {
-            InitialValidator memory initialValidator = subnetConversionData.initialValidators[i];
-            encodedConversion = abi.encodePacked(
-                encodedConversion,
-                initialValidator.nodeID,
-                initialValidator.weight,
-                initialValidator.blsPublicKey
-            );
-        }
-        require(
-            sha256(encodedConversion) == subnetConversionID,
-            "ValidatorManager: invalid subnet conversion ID"
-        );
         require(
             subnetConversionData.blockchainID == WARP_MESSENGER.getBlockchainID(),
             "ValidatorManager: invalid blockchain ID"
         );
-
-        // TODO: might have more efficient conversion with assembly, check length of bytes
         require(
-            subnetConversionData.validatorManagerAddress.length == 20,
-            "ValidatorManager: invalid validator manager address length"
-        );
-        require(
-            address(bytes20(subnetConversionData.validatorManagerAddress)) == address(this),
+            address(subnetConversionData.validatorManagerAddress) == address(this),
             "ValidatorManager: invalid validator manager address"
         );
         require(
             subnetConversionData.initialValidators.length > 0,
             "ValidatorManager: no initial validators"
         );
+        uint256 numInitialValidators = subnetConversionData.initialValidators.length;
+        require(
+            numInitialValidators < type(uint32).max, "ValidatorManager: too many initial validators"
+        );
 
-        // Insert the initial validators into the storage.
+        // Verify that the sha256 hash of the Subnet conversion data matches with the Warp message's subnetConversionID.
+        bytes memory encodedConversion = abi.encodePacked(
+            subnetConversionData.convertSubnetTxID,
+            subnetConversionData.blockchainID,
+            uint32(20),
```

Comment:
ah I see that you are hardcoding length here instead of doing the bytes packing, I'm not sure what is cleaner long term